pkg/crc/network/proxy.go

```diff
@@ -0,0 +1,201 @@
+package network
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+)
+
+var defaultNoProxies = []string{".apps-crc.testing", "127.0.0.1", "api.crc.testing", "localhost"}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters. If a empty string is passed
+// the corresponding environment variable is checked.
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	defaultScheme := "http"
+
+	// We only parse the Proxy Special char if it is coming from the config file
+	// or user defined flag. In case of Env variable we just ignore parsing.
+	if httpProxy != "" {
+		httpProxy = parseProxySpecialChar(httpProxy, defaultScheme)
+	}
+	if httpsProxy != "" {
+		httpsProxy = parseProxySpecialChar(httpsProxy, defaultScheme)
+	}
+
+	if httpProxy == "" {
+		httpProxy = os.Getenv("http_proxy")
+		if httpProxy == "" {
+			httpProxy = os.Getenv("HTTP_PROXY")
+		}
+	}
+	err := ValidateProxyURL(httpProxy, defaultScheme)
```

Comment:
govalidator.IsURL is provide a better way to parse a url to check it validity, but since we already removing `parseProxySpecialChar()` one so that should be not an issue.